### ISSUE TITLE
Rank to MMR

### ISSUE THREAD

**Issue by dankYoff** [dotabod/backend#396]
I wrote a code that can convert rank to mmr. To do this, we substitute the rank in the 1st or 2nd formula (depending on the rank). And we can see how the rank and MMR correlate. For high ranks, the error is 100-300 MMR (> rank =  <accuracy: it depends on the top, as well as a limited amount of information about the ranks, I had 15 of them). 

But in general, this can be easily corrected and the formulas can be made even more accurate, although they are already quite accurate in calculating MMR.

### MAIN
Accuracy: 95-98% | If Rank <=460
Formula: MMR = -448.45658 * ln(9264.01245 * "RANK") + 17242.22337

Accuracy: 98-99% | If Rank>460
Formula: MMR = -1248.02749 * ln(1861.20536 * "RANK") + 27052.29973

### Example:
Rank 131 | ~ 11000-11150 MMR
MMR = -448.45658 * ln(9264.01245 * 131) + 17242.22337 = 10959.75 MMR
Accuracy: 98.3% 

Rank 1674 | ~ 8245 MMR
MMR = -1248.02749 * ln(1861.20536 * 1674) + 27052.29973 = 8391 MMR
Accuracy: 98.2% 

### Image for example: 
![image](https://github.com/dotabod/backend/assets/91318807/3f426227-5e00-4a10-9008-d1c6ad2b8763)

**Comment by Geczy** [comment]
is this EU only or does it hold true for russia, eu, sea, us east, us west?

awesome job with the formulas!

**Comment by dankYoff** [comment]
I don't think it makes sense to do this for the US/SEA, because there are more streams from either the CIS or Europe, and I'm not sure if it's possible to track the region of the match.

The rating systems are different everywhere, because the number of players varies from region to region.

**Comment by dankYoff** [comment]
### _**How to do this for any server.**_ 
Step 1. We have data from dotabod: what mmr = what rank. So we can do the following. We take people from the database, look at which server they played the last match on, and write them into the database. Now we know and can calculate the average rating for each region. 

Step 2. From each region CIS/Europe/Chine/US West|East, we select a separate region and calculate the rank approximation to mmr, as it was in my code, the more data the more accurate the result will be.

Step 3. We get formulas that now calculate the average player rating at any given time.

Ez

**Comment by Geczy** [comment]
thats a good idea @dankYoff 

looks like opendota returns the region in api https://docs.opendota.com/#tag/matches/operation/get_matches_by_match_id
maybe so does steam api, but i haven't checked

**Comment by dankYoff** [comment]
Code: https://github.com/dankYoff/rank-to-mmr/blob/main/VHS.py

It recalculates the coefficients for both MMRs (< and > 460 rank),

Why 460? Because we have 2 graphs, first one curve and then a sharply different one. Perhaps the situation has changed and we need to change the rank and formula of the graphs, but the idea with the rank correlation is the best I have come up with

![Image](https://github.com/user-attachments/assets/9e37e763-2751-4a60-8f60-7860f91b6826)
